homeassistant/components/tuya/entity.py

```diff
@@ -47,6 +47,11 @@ def step_scaled(self) -> float:
 
     def scale_value(self, value: float) -> float:
         """Scale a value."""
+        if isinstance(value, str):
+            try:
+                value = float(value)
+            except ValueError:
+                value = 0
```

Comment:
This seems incorrect. If we are not getting a float value, that would be unexpected. An entity can fallback to an Unknown state at that point. Setting it to 0 is a huge side effect (for example, if it is a every increasing counter, you have resetted it by doing this).

Reply:
I understand this, I'm okay with not implementing this change. I will be happy to revert this.
The reason I implemented this was, the function expects a float as an input `def scale_value(self, value: float) -> float:`
If the value is not a float as a string, we can rather than resetting to 0, return the actual value if it does not convert to a float.
Let me know your thoughts.

Reply:
My thoughts I already expressed. This is not a correct behavior.

Reply:
Understood.

Reply:
This change has been reverted.